frontend_new/src/components/BookmarkDrawer.tsx

```diff
@@ -1,87 +1,199 @@
+import Grid from "@material-ui/core/Grid";
+import SimpleCard from "./Cards.tsx";
 
-const useStyles = makeStyles((theme) => ({
-  root: {
-    flexGrow: 1,
-  },
-  menuButton: {
-    marginRight: theme.spacing(2),
-  },
-  title: {
-    flexGrow: 1,
-  },
-  list: {
-    width: 300,
-  },
-  fullList: {
-    width: "auto",
-  },
-}));
+const drawerWidth = "100%";
 
-type Anchor = "bookmark";
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      display: "flex",
+    },
+    appBar: {
+      transition: theme.transitions.create(["margin", "width"], {
+        easing: theme.transitions.easing.sharp,
+        duration: theme.transitions.duration.leavingScreen,
+      }),
+      zIndex: theme.zIndex.drawer + 1,
+    },
+    appBarShift: {
+      width: `calc(100% - ${drawerWidth}px)`,
+      marginLeft: drawerWidth,
+      transition: theme.transitions.create(["margin", "width"], {
+        easing: theme.transitions.easing.easeOut,
+        duration: theme.transitions.duration.enteringScreen,
+      }),
+    },
+    menuButton: {
+      marginRight: theme.spacing(2),
+    },
+    hide: {
+      display: "none",
+    },
+    drawer: {
+      flexShrink: 0,
+    },
+    drawerPaper: {
+      width: drawerWidth,
+      backgroundColor: theme.palette.background.default,
+      height: "100%",
+      overflowX: "hidden",
+    },
+    drawerHeader: {
+      display: "flex",
+      alignItems: "center",
+      padding: theme.spacing(0, 1),
+      // necessary for content to be below app bar
+      ...theme.mixins.toolbar,
+      justifyContent: "flex-end",
+    },
+    content: {
+      flexGrow: 1,
+      padding: theme.spacing(3),
+      transition: theme.transitions.create("margin", {
+        easing: theme.transitions.easing.sharp,
+        duration: theme.transitions.duration.leavingScreen,
+      }),
+      marginLeft: -drawerWidth,
+    },
+    contentShift: {
+      transition: theme.transitions.create("margin", {
+        easing: theme.transitions.easing.easeOut,
+        duration: theme.transitions.duration.enteringScreen,
+      }),
+      marginLeft: 0,
+    },
+  })
+);
 type props = {
-  whichTheme: boolean;
+  bookmarks: array;
+  handlebookmark: () => void;
+  whichTheme: Theme;
+  themeChoice: () => void;
 };
 
-export default function BookmarkDrawer(props: props) {
+export default function PersistentDrawerLeft(props: props) {
   const classes = useStyles();
-  const [state, setState] = React.useState({
-    bookmark: false,
-  });
+  const theme = useTheme();
+  const [open, setOpen] = React.useState(false);
 
-  const toggleDrawer =
-    (anchor: Anchor, open: boolean) =>
-    (event: React.KeyboardEvent | React.MouseEvent) => {
-      if (
-        event.type === "keydown" &&
-        ((event as React.KeyboardEvent).key === "Tab" ||
-          (event as React.KeyboardEvent).key === "Shift")
-      ) {
-        return;
-      }
+  const handleDrawerOpen = () => {
```

Comment:
**[Question]** Why do we need two separate functions to handle?

Reply:
Yeah could defs combine them. Took the drawer implementation off mui website

Reply:
I dont necessarily think combining the functions is necessary